js/ui/components/clubhouse.js

```diff
@@ -534,6 +676,9 @@ var ClubhouseComponent = new Lang.Class({
                     this._clearQuestBanner();
                 }
             });
+
+            let clubhouseAnimator = getClubhouseAnimator();
+            clubhouseAnimator.setClubhouseProxy(this.proxy);
```

Comment:
It's clear that the proxy is really a construct parameter for the animator... I understand that it's a singleton, but instead of pretending that it does not matter who calls `setClubhouseProxy()` I think I would find it cleaner if we had an e.g. `createClubhouseAnimator()` method that explicitly takes the proxy as a construct parameter, which would only be called by the component.

Reply:
I changed this to a more natural flow.